phonenumbers.go

```diff
@@ -2580,12 +2579,11 @@ func maybeExtractCountryCode(
 			// if it was too long before, we consider the number with
 			// the country calling code stripped to be a better result and
 			// keep that instead.
-			finds := validNumberPattern.FindAllString(fullNumber.String(), -1)
+			fullValid := validNumberPattern.MatchString(fullNumber.String())
+			nationalValid := validNumberPattern.MatchString(potentialNationalNumber.String())
+			lengthValid := testNumberLength(fullNumber.String(), defaultRegionMetadata, UNKNOWN)
 
-			cond := (len(finds) != 0 && fullNumber.String() == finds[0] &&
-				validNumberPattern.MatchString(potentialNationalNumber.String())) ||
-				testNumberLength(fullNumber.String(), defaultRegionMetadata, UNKNOWN) == TOO_LONG
-			if cond {
+			if (!fullValid && nationalValid) || lengthValid == TOO_LONG {
```

Comment:
Note `patP` above takes care of adding anchors for start and end, so `MatchString` here is right.